functionary/prompt_template/base_template.py

```diff
@@ -261,27 +261,43 @@ def get_raw_response_from_assistant_message(
         for stop_token in self.get_stop_tokens_for_generation():
             raw_response = raw_response.replace(stop_token, "")
 
-        if tool_func_choice == "none":
-            raw_response = raw_response[len(self.get_force_text_generation_prefix()) :]
-        elif tool_func_choice == "required":
-            raw_response = raw_response[len(self.get_tool_choice_required_prefix()) :]
-        elif isinstance(tool_func_choice, Tool) or isinstance(
-            tool_func_choice, Function
-        ):
-            raw_response = raw_response[
-                len(
-                    self.get_force_function_call_prefix(
-                        function_name=default_tool_call_name
-                    )
-                ) :
-            ]
+        generation_prefix = self.get_generation_prefix_for_tool_choice(tool_func_choice)
+        raw_response = raw_response[len(generation_prefix) :]
+        # if tool_func_choice == "none":
+        #     raw_response = raw_response[len(self.get_force_text_generation_prefix()) :]
+        # elif tool_func_choice == "required":
+        #     raw_response = raw_response[len(self.get_tool_choice_required_prefix()) :]
+        # elif isinstance(tool_func_choice, Tool) or isinstance(
+        #     tool_func_choice, Function
+        # ):
+        #     raw_response = raw_response[
+        #         len(
+        #             self.get_force_function_call_prefix(
+        #                 function_name=default_tool_call_name
+        #             )
+        #         ) :
+        #     ]
```

Comment:
Remove comments

Reply:
Ok, here is your old code, I just kept it as comment for you to check it easily